052-cruise-control-api-users.md

```diff
@@ -0,0 +1,262 @@
+
+Cruise Control's authorization credential files adhere to Jetty's HashLoginService's file format: `username: password [,rolename ...]`.
+
+By default Cruise Control defines three roles: `VIEWER`, `USER` and `ADMIN`.
+```
+VIEWER role: has access to the most lightweight kafka_cluster_state, user_tasks and review_board endpoints.
+USER role: has access to all the GET endpoints except bootstrap and train.
+ADMIN role: has access to all endpoints.
+```
+This proposal is for supporting the `USER` and `VIEWER` roles only.
+For more details on these roles checkout the [Cruise Control Wiki](https://github.com/linkedin/cruise-control/wiki/Security#authorization)
+
+So a developer could create a Cruise Control API authentication credentials file named `cruise-control-auth.txt`, with the following content:
+```
+userOne: passwordOne, USER
+userTwo: passwordTwo, VIEWER
+```
+
+Then the developer can use this file to create a secret using the following command:
+```
+kubectl create secret generic cruise-control-api-users-secret  --from-file=key=cruise-control-auth.txt
+```
+
+The command would create a secret:
+```yaml=
+apiVersion: v1
+kind: Secret
+metadata:
+  name: cruise-control-api-users-secret
+type: Opaque
+data:
+  key: dXNlck9uZTogcGFzc3dvcmRPbmUsIFVTRVIKdXNlclR3bzogcGFzc3dvcmRUd28sIFZJRVdFUgo=
+```
+
+The developer could then reference the secret in a `spec.cruiseControl.apiUsers` section of the `Kafka` resource.
+
+The schema would look like this:
+
+```yaml=
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  ...
+spec:
+  cruiseControl:
+    apiUsers:
+      type: fromSecret (1)
+      secret:
+        secretName: cruise-control-api-users-secret
+        secretKey: key
+     ...
+```
+(1) A type field is added here for configuration flexibility in the future. e.g. Gives us the option to have other methids of configuring API users other than using `secrets`
+
+Strimzi would decode and use the contents of this secret to populate the Cruise Control API auth credentials file which would be mounted and used by the Cruise Control pod.
+
+Although this would simplify the overall design it would also shift a lot of the maintanence responsibility to the developer.
+The developer would need to create a valid Cruise Control API user configuration and encode it into a secret.
+That being said, this is still a reasonable amount of responsibility to leave to the developer.
+This is because accessing the Cruise Control API directly is not the primary way of interacting with Cruise Control for a Strimzi-managed Kafka cluster and is expected to be used only by advanced developers in special use-cases.
+However, with proper supporting documentation, this new feature will give developers a greater degree of control and assistance with special use-cases.
+
+### Risks
+
+Supporting developers to create Cruise Control API users with non-`ADMIN` roles like `USER` and `VIEWER` roles would mitigate this risk completely.
```

Comment:
Is this missing some previous sentence that defines the `this risk`? Or maybe you should instead of `this risk` reference the risks described int he _Motivation_ section.